packages/Select/src/Select.js

```diff
@@ -1,94 +1,139 @@
-    { "form-select--is-readonly": isReadOnly },
-    { "form-select--has-error": hasError },
-    className
-  );
+  const renderPlaceholder = () => {
+    if (!placeholder) return null;
+    return (
+      <option disabled value="" data-pka-anchor="select.placeholder">
+        {placeholder}
+      </option>
+    );
+  };
+
+  function getDefaultValue() {
+    if (isControlled) return undefined;
+    if (defaultValue === null && !!placeholder) return "";
+    return defaultValue;
+  }
 
   return (
-    <sc.Select className={rootClasses} data-pka-anchor="select">
-      <select
+    <sc.SelectContainer data-pka-anchor="select.container" {...containerProps}>
+      <sc.Select
         aria-invalid={hasError}
-        className="form-select__select"
+        aria-readonly={isReadOnly}
+        aria-label={a11yText}
+        data-pka-anchor="select"
+        defaultValue={getDefaultValue()}
         disabled={isDisabled || isReadOnly}
-        ref={ref}
+        onChange={callAll(handleChange, onChange)}
+        ref={refSelect}
         value={value}
+        {...styleProps}
         {...moreProps}
       >
         {renderPlaceholder()}
         {children}
-      </select>
-    </sc.Select>
+      </sc.Select>
+      <sc.CaretIcon data-pka-anchor="select.arrow-icon" isDisabled={isDisabled || isReadOnly} />
+    </sc.SelectContainer>
   );
 });
 
 Select.types = {
   size: constants.defaultSize,
 };
 
-const propTypes = {
+Select.propTypes = {
+  /** Provides a non-visible label for this select element for assistive technologies. */
   a11yText: PropTypes.string,
-  className: PropTypes.string,
+
+  /** List of options as standard option elements. */
   children: PropTypes.node,
+
+  /** Sets the default selected value for an uncontrolled component. */
+  defaultValue: PropTypes.string,
+
+  /** If true displays a red border around select element to indicate error. */
   hasError: PropTypes.bool,
+
+  /** If true it makes the select element disabled. */
   isDisabled: PropTypes.bool,
+
+  /** If true it makes the select element read only. */
   isReadOnly: PropTypes.bool,
+
+  /** Callback to be executed when the selected value is changed. Receives the onChange event as an argument.  Required when value prop is provided (component is controlled). */
+  onChange: PropTypes.func,
+
+  /** Display value for a disabled first option with an empty string value. */
   placeholder: PropTypes.string,
+
+  /** Specifies the visual size of the select element. */
   size: PropTypes.oneOf([Select.types.size.SMALL, Select.types.size.MEDIUM, Select.types.size.LARGE]),
+
+  /** The selected value for the select element. */
   value: PropTypes.string,
 };
 
-const defaultProps = {
+Select.defaultProps = {
   a11yText: null,
-  className: null,
   children: null,
+  defaultValue: null,
   hasError: false,
   isDisabled: false,
   isReadOnly: false,
+  onChange: () => {},
   placeholder: null,
   size: Select.types.size.MEDIUM,
-  value: "",
+  value: undefined,
```

Comment:
Does `null` work? all the other default empty values are `null`

Reply:
If you pass `value=null` to the underlying `<select>` React thinks you want it to be a controlled component, and it stops working properly. I think. I'll double check though.

Reply:
oh I didn't know that, thanks

Reply:
Actually it does still work, but there are console warnings from React.

```
Warning: `value` prop on `select` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
```

and

```
Warning: Select elements must be either controlled or uncontrolled (specify either the value prop, or the defaultValue prop, but not both). Decide between using a controlled or uncontrolled select element and remove one of these props.
```